openshift-client/src/main/java/io/fabric8/openshift/client/dsl/internal/OpenShiftOperation.java

```diff
@@ -63,18 +63,15 @@ public OpenShiftConfig getConfig() {
 
   @Override
   public URL getRootUrl() {
-    OpenShiftConfig config = OpenShiftConfig.wrap(context.getConfig());
-    OpenShiftClient oc = new DefaultOpenShiftClient(context.getClient(), config);
-    if (config.isOpenShiftAPIGroups(oc)) {
-      oc.close();
-      return super.getRootUrl();
-    } else {
-      oc.close();
+    // This is an OpenShift resource. If no API Group Name is specified, use /oapi endpoint
+    if (Utils.isNullOrEmpty(context.getApiGroupName())) {
```

Comment:
Thanks, seems fine to me. Have you tested this on both OS3 and OS4?

Reply:
@rohanKanojia Yes, I tested in both OS3 and OS4 and it works great! :smile: